recipes/webhooks_example.ipynb

```diff
@@ -0,0 +1,522 @@
+    "from typing import Dict\n",
+    "\n",
+    "def check_rules_on_label(label: Dict):\n",
+    "    #custom methods \n",
+    "    print('Custom method - checking number of bboxes')\n",
+    "\n",
+    "    counter = 0\n",
+    "    for annotation in label['jsonResponse'][\"JOB_0\"][\"annotations\"]:\n",
+    "        if annotation[\"categories\"][0][\"name\"] == \"OBJECT_A\":\n",
+    "            counter += 1\n",
+    "\n",
+    "    if counter == 0:\n",
+    "        return []\n",
+    "    return [f'There are too many BBox ({counter}) - Only 1 BBox of Object A accepted']\n",
+    "\n",
+    "class PluginHandler(PluginCore):\n",
+    "    \"\"\"\n",
+    "    Custom plugin instance\n",
+    "    \"\"\"\n",
+    "\n",
+    "    def on_submit(self, label: Dict, asset_id: str) -> None:\n",
+    "        \"\"\"\n",
+    "        Dedicated handler for Submit action \n",
+    "        \"\"\"\n",
+    "        self.logger.info(\"On submit called\")\n",
+    "\n",
+    "        issues_array = check_rules_on_label(label)\n",
+    "\n",
+    "        project_id = self.project_id\n",
+    "\n",
+    "        if len(issues_array) > 0:\n",
+    "            print(\"Creating an issue...\")\n",
+    "\n",
+    "            for i, _ in enumerate(issues_array):\n",
+    "\n",
+    "                self.kili.append_to_issues(\n",
+    "                    label_id=label['id'],\n",
+    "                    project_id=project_id,\n",
+    "                    text=issues_array[i],\n",
+    "                )\n",
+    "\n",
+    "            print(\"Issue created!\")\n",
+    "\n",
+    "            self.kili.send_back_to_queue(asset_ids=[asset_id])\n",
+    "\n",
+    "```"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "57b991ca",
+   "metadata": {},
+   "source": [
+    "You will need to deploy this on your premise for this to work. Easy solutions are [FastAPI](https://fastapi.tiangolo.com/), with a few lines of codes.\n",
+    "For this demo, we will use https://webhook.site that will allow us to explore the payload of the calls.\n",
+    "\n",
+    "```python\n",
+    "# file main.py\n",
+    "from fastapi import FastAPI\n",
+    "from .plugin import PluginHandler\n",
+    "\n",
+    "app = FastAPI()\n",
+    "kili = Kili()\n",
+    "\n",
+    "@app.get(\"/\")\n",
+    "def main(raw_payload):\n",
+    "    event_type = raw_payload.get('eventType')\n",
+    "    project_id = raw_payload.get('logPayload').get('projectId')\n",
+    "\n",
+    "    if (not project_id):\n",
+    "        print('Invalid projectId')\n",
+    "        return\n",
+    "    \n",
+    "    project_id = payload.get('projectId')\n",
+    "    plugin = PluginHandler(kili, project_id)\n",
+    "\n",
+    "    if (not event_type):\n",
+    "        print('Invalid event')\n",
+    "        return\n",
+    "\n",
+    "    payload = raw_payload.get('payload')\n",
+    "    label = payload.get('label')\n",
+    "    asset_id = payload.get('assetId')\n",
```

Comment:
Should be asset_id, no ?

Reply:
Nope, we have camel case in the event's payload

Reply:
It's camelCase in the raw_payload, but snake case for the `payload` :

<img width="993" alt="image" src="https://user-images.githubusercontent.com/99664787/222773371-f55514fc-4ae3-4d0f-9113-2c54ac2ad25b.png">

Reply:
Not really consistent, but it was done to automatically call the good handler for the plugins